src/Binding/AbstractBinding.php

```diff
@@ -138,39 +181,43 @@ public function hasParameterValue($parameterName)
     /**
      * {@inheritdoc}
      */
-    public function equals(ResourceBinding $other)
+    public function serialize()
     {
-        if (get_class($other) !== get_class($this)) {
-            return false;
-        }
+        $data = array();
 
-        if ($this->query !== $other->getQuery()) {
-            return false;
-        }
+        $this->preSerialize($data);
 
-        if ($this->type !== $other->getType()) {
-            return false;
-        }
+        return serialize($data);
+    }
 
-        if ($this->language !== $other->getLanguage()) {
-            return false;
-        }
+    /**
+     * {@inheritdoc}
+     */
+    public function unserialize($serialized)
+    {
+        $data = unserialize($serialized);
 
-        // The local parameters are sorted by key. Sort before comparing to
-        // prevent false negatives.
-        $otherParameterValues = $other->getParameterValues();
-        ksort($otherParameterValues);
+        $this->postUnserialize($data);
+    }
 
-        if ($this->parameterValues !== $otherParameterValues) {
-            return false;
-        }
+    protected function preSerialize(array &$data)
+    {
+        $data[] = $this->typeName;
+        $data[] = $this->userParameterValues;
+        $data[] = $this->uuid->toString();
+    }
 
-        return true;
+    protected function postUnserialize(array $data)
+    {
+        $this->uuid = Uuid::fromString(array_pop($data));
```

Comment:
inlining the call will not solve the issue when subclasses are overwriting the method though, as they will still have a ``$data`` variable holding a reference while they call the parent method.